packages/react-scripts/backpack-addons/splitChunks.js

```diff
@@ -4,23 +4,34 @@ const paths = require('../config/paths');
 const appPackageJson = require(paths.appPackageJson);
 const bpkReactScriptsConfig = appPackageJson['backpack-react-scripts'] || {};
 
-module.exports = (isEnvDevelopment) => {
+module.exports = isEnvDevelopment => {
   // Automatically split vendor and commons
   // https://twitter.com/wSokra/status/969633336732905474
   // https://medium.com/webpack/webpack-4-code-splitting-chunk-graph-and-the-splitchunks-optimization-be739a861366
   return {
     splitChunks: bpkReactScriptsConfig.enableAutomaticChunking
-    ? {
-      chunks: 'all',
-      name: isEnvDevelopment,
-      cacheGroups: bpkReactScriptsConfig.vendorsChunkRegex
+      ? isEnvDevelopment
         ? {
-            defaultVendors: {
-              test: new RegExp(bpkReactScriptsConfig.vendorsChunkRegex)
-            },
+            chunks: 'all',
+            cacheGroups: bpkReactScriptsConfig.vendorsChunkRegex
+              ? {
+                  defaultVendors: {
+                    test: new RegExp(bpkReactScriptsConfig.vendorsChunkRegex),
+                  },
+                }
+              : {},
           }
-        : {},
-      }
-    : {}
-  }
+        : {
+            chunks: 'all',
+            name: false,
+            cacheGroups: bpkReactScriptsConfig.vendorsChunkRegex
+              ? {
+                  defaultVendors: {
+                    test: new RegExp(bpkReactScriptsConfig.vendorsChunkRegex),
+                  },
+                }
+              : {},
+          }
+      : {},
+  };
```

Comment:
Good to use some common code to implement this functionality, no need to have a deep structure of code.